addons/sourcemod/scripting/gokz-replays/playback.sp

```diff
@@ -937,12 +938,15 @@ void PlaybackVersion2(int client, int bot, int &buttons)
 			EmitSoundToClientSpectators(client, gC_ModeEndSounds[GOKZ_GetCoreOption(client, Option_Mode)]);
 		}
 
-		// Set velocity to travel from current origin to recorded origin
-		float currentOrigin[3], velocity[3];
-		Movement_GetOrigin(client, currentOrigin);
-		MakeVectorFromPoints(currentOrigin, currentTickData.origin, velocity);
-		ScaleVector(velocity, 1.0 / GetTickInterval());
-		TeleportEntity(client, NULL_VECTOR, currentTickData.angles, velocity);
+		TeleportEntity(client, NULL_VECTOR, currentTickData.angles, currentTickData.velocity);
```

Comment:
Calculating it from the origin will fail if the player gets teleported as the velocity becomes inaccurate, causing weird interpolation issues and/or sound issues

I was unaware of this issue until now, what happened previously?

Reply:
If I remember correctly, the bot would start to drift off in some cases, I assume due to rounding differences, but it's been a hot minute since we last tried that. Would the interpolation issues be solved if we only took the replay velocity on teleport ticks? We can directly set the origin as well in those cases if that'd help I would say.

Reply:
As the origin of the bot gets updated every tick, drifting should not happen.
About taking velocity during teleport ticks only, this might cause issues if you get teleported by the maps themselves by a short distance as those aren't recorded into the replays themselves, and as far as the client is concerned, being teleported by a map or by the menu should be the same to them.

Reply:
teleported means bot's velocity becoms very high, why not just teleport them at that tick?